src/java/org/apache/cassandra/streaming/StreamSession.java

```diff
@@ -542,7 +542,9 @@ private Future<?> closeSession(State finalState, String failureReason)
                     this.failureReason = failureReason;
 
                     sink.onClose(peer);
-                    streamResult.handleSessionComplete(this);
+                    // closed before init?
```

Comment:
Looks like it... `streamOperation()` certainly assumes that's the state of things if it sees a `null` `streamResult`.

Reply:
left this comment here by intent, so next author knows why its there